cloud: update branch limitations (#22389)
pingcap/docs: tidb-cloud/branch-overview.md

- For each branch of a free cluster, 10 GiB storage is allowed. For each branch of a cluster with spending limit > 0, 100 GiB storage is allowed. Once the storage is reached, the read and write operations on this branch will be throttled until you reduce the storage.

> - Branches are intended for short-term feature development and functional testing. Because branches lack auto-scaling capability, performance testing is not supported.

Reviewer: ```suggestion
- Branches are intended for short-term feature development and functional testing. Because branches lack auto-scaling capability, they are not suitable for performance testing.
```
